lib/internal/Magento/Framework/Cache/Frontend/Decorator/TagScope.php

```diff
@@ -61,10 +61,8 @@ public function clean($mode = \Zend_Cache::CLEANING_MODE_ALL, array $tags = [])
     {
         if ($mode == \Zend_Cache::CLEANING_MODE_MATCHING_ANY_TAG) {
             $result = false;
-            foreach ($tags as $tag) {
-                if (parent::clean(\Zend_Cache::CLEANING_MODE_MATCHING_TAG, [$tag, $this->getTag()])) {
-                    $result = true;
-                }
+            if (parent::clean(\Zend_Cache::CLEANING_MODE_MATCHING_TAG, array_merge($tags, [$this->getTag()]))) {
```

Comment:
Hi @raivisdejus,
Unfortunately it will not work, as it mean that will be cleaned entities that contains all listed tags, which is incorrect. Could you fix it? 

Also please review and fix failing tests.

Reply:
True, adjusted to `\Zend_Cache::CLEANING_MODE_MATCHING_ANY_TAG` will check tests as well. Thanks.